### ISSUE TITLE
HighDPI issues on linux

### ISSUE THREAD

**Issue by mgerhardy** [wolfpld/tracy#65]
it's not possible to use the profiler on my laptop's screen resolution (3840x2160).

**Comment by mgerhardy** [comment]
![Screenshot from 2020-07-02 16-26-19](https://user-images.githubusercontent.com/577713/86371130-d692c480-bc80-11ea-909d-fe9be064fc3e.png)

**Comment by mgerhardy** [comment]
![Screenshot from 2020-07-02 16-27-10](https://user-images.githubusercontent.com/577713/86371208-eb6f5800-bc80-11ea-8fb7-b9406c6b0170.png)

**Comment by wolfpld** [comment]
You can set an appropriate `dpiScale` in `profiler/src/main.cpp`. Ideally this should be retrieved from the system.

**Comment by mgerhardy** [comment]
Shouldn't `glfwGetMonitorContentScale` help here cross platform? According to the docs it should:

* [glfw-docs](https://www.glfw.org/docs/latest/group__monitor.html#gad3152e84465fa620b601265ebfcdb21b)

Will give it a try and let you know.

**Comment by mgerhardy** [comment]
Something like this - unfortunately I can't test this as I currently only have glfw 3.2 - I first to build a newer version.

```
diff --git a/profiler/src/main.cpp b/profiler/src/main.cpp
index 2485ead7..73ddedc1 100644
--- a/profiler/src/main.cpp
+++ b/profiler/src/main.cpp
@@ -242,6 +242,14 @@ int main( int argc, char** argv )
     gl3wInit();
     glfwSetWindowRefreshCallback( window, WindowRefreshCallback );
 
+#if GLFW_VERSION_MAJOR > 3 || (GLFW_VERSION_MAJOR == 3 && GLFW_VERSION_MINOR >= 3)
+    GLFWmonitor* monitor = glfwGetWindowMonitor(window);
+    if (monitor != nullptr) {
+        float xscale, yscale;
+        glfwGetMonitorContentScale(monitor, &xscale, &yscale);
+        dpiScale = xscale > yscale : xscale : yscale;
+    }
+#else
 #ifdef _WIN32
     typedef UINT(*GDFS)(void);
     GDFS getDpiForSystem = nullptr;
@@ -250,6 +258,7 @@ int main( int argc, char** argv )
         getDpiForSystem = (GDFS)GetProcAddress(dll, "GetDpiForSystem");
     if (getDpiForSystem)
         dpiScale = getDpiForSystem() / 96.f;
+#endif
 #endif
 
     // Setup ImGui binding

```

**Comment by mgerhardy** [comment]
tested the above now - works quite nice now. Not sure which version of glfw you are targeting for. But if 3.3 is the min version - one can also remove the checks and the win32 code.

**Comment by mgerhardy** [comment]
I suppose I have to add the `glfwGetPrimaryMonitor` call, because the window is not yet initialized properly at that point. I think it would be better to move the dpiScale assignment after the window init. Looks like creating the window is not enough here.

**Comment by wolfpld** [comment]
`glfwGetWindowMonitor` doesn't do what you think it does (see docs). You'd need to enumerate available monitors (`glfwGetMonitors`) to detect on which monitor the window will be displayed, taking into account the possibility that monitor configuration might have changed since last run, and the saved window coordinates could have been adjusted by the window manager (or not at all, placing the window out-of-view, who knows).

Then you have trouble with multiple monitors with different DPI settings, but this problem is already present right now (not on Windows, which scale the window-contents bitmap to match the target monitor DPI). Proper fix would be needed to be done on ImGui side, I believe @ocornut has it in the works.

**Comment by mgerhardy** [comment]
But it should at least do the (almost) same as the win32 specific code, no? I know that it doesn't deal with the edge cases.

At least it fixes the problem for me in a way, that renders the profiler's ui readable.

**Comment by mgerhardy** [comment]
Or maybe provide a way to override the dpi scale factor with an env var? at least as a workaround...

**Comment by wolfpld** [comment]
Implemented as proposed in 5abd4c8a.
